app/assets/javascripts/index/directions.js

```diff
@@ -216,6 +216,8 @@ OSM.Directions = function (map) {
 
     endpoints[0].enableListeners();
     endpoints[1].enableListeners();
+
+    endpoints[0].focusInput();
```

Comment:
Well then how about focusing only when reasonable?
```suggestion
    if (!endpoints[0].value) endpoints[0].focusInput();
```